### ISSUE TITLE
Can't save with nested array in blocks

### ISSUE THREAD

**Issue by danny1ng** [payloadcms/payload#4155]
### Describe the Bug

Hello.

I have found trouble connected to double depth of the array in blocks. 
When I try to save, I get errors

![image](https://github.com/payloadcms/payload/assets/59068428/a4996eae-2820-43c2-9ebc-ee4cc9b58d6b)
![image](https://github.com/payloadcms/payload/assets/59068428/cd0853ad-1602-4c03-bcfe-f5b2a8b10488)

### To Reproduce

1. Just add this code to collections
```
import type { Block, CollectionConfig, Field } from 'payload/types';

const columnFields: Field[] = [
  {
    name: 'text',
    required: true,
    type: 'text',
  },
  {
    name: 'include',
    required: true,
    type: 'checkbox',
    defaultValue: true,
  },
];

export const PriceListBlock: Block = {
  fields: [
    {
      name: 'priceCard',
      fields: [
        {
          name: 'title',
          required: true,
          type: 'text',
        },
        {
          name: 'planList',
          fields: columnFields,
          type: 'array',
        },
      ],
      type: 'array',
    },
  ],
  slug: 'priceListTest',
};

export const TestCollection: CollectionConfig = {
  admin: {
    defaultColumns: ['title', 'slug', 'updatedAt'],
  },
  fields: [
    {
      name: 'title',
      required: true,
      type: 'text',
    },

    {
      name: 'layout',
      blocks: [PriceListBlock],
      type: 'blocks',
    },
  ],

  slug: 'TestCollection',
  versions: {
    drafts: false,
  },
};
```

2. Add  block layout priceList and fill in all the fields.

### Payload Version

2.1.1

### Adapters and Plugins

db-postgres

**Comment by kyuumeitai** [comment]
It happens to me as well. Given that your error is an image, I'll paste text to others (like me) can find it through search.

```
 [14:25:55] ERROR (payload): There was an error while saving a version for the Survey with ID 1.
 [14:25:55] ERROR (payload): null value in column "_parent_id" violates not-null constraint
     err: {
       "type": "DatabaseError",
       "message": "null value in column \"_parent_id\" violates not-null constraint",
       "stack":
           error: null value in column "_parent_id" violates not-null constraint
               at /home/node/app/node_modules/pg/lib/client.js:526:17
               at processTicksAndRejections (node:internal/process/task_queues:95:5)
               at async /home/node/app/node_modules/src/node-postgres/session.ts:64:19
               at async insertArrays (/home/node/app/node_modules/@payloadcms/db-postgres/src/upsertRow/insertArrays.ts:65:7)
               at async insertArrays (/home/node/app/node_modules/@payloadcms/db-postgres/src/upsertRow/insertArrays.ts:75:7)
               at async upsertRow (/home/node/app/node_modules/@payloadcms/db-postgres/src/upsertRow/index.ts:258:5)
               at async Object.createVersion (/home/node/app/node_modules/@payloadcms/db-postgres/src/createVersion.ts:27:18)
               at async saveVersion (/home/node/app/node_modules/payload/src/versions/saveVersion.ts:100:18)
               at async updateByID (/home/node/app/node_modules/payload/src/collections/operations/updateByID.ts:276:16)
               at async updateByIDHandler (/home/node/app/node_modules/payload/src/collections/requestHandlers/updateByID.ts:36:17)
       "length": 276,
       "name": "error",
       "severity": "ERROR",
       "code": "23502",
       "detail": "Failing row contains (1, null, 4, mejor otro, a, null, nop, 655cbd62c538ca66b49e2cd0).",
       "schema": "public",
       "table": "_surveys_v_version_questions_alternatives",
       "column": "_parent_id",
       "file": "execMain.c",
       "line": "2032",
       "routine": "ExecConstraints"
     }
     
     ```

"payload": "^2.2.0",
"@payloadcms/db-postgres": "^0.1.13",

**Comment by ixoye36** [comment]
Hi

I face exactly the same issue for the same use case

Here is my pricing block to hold my pricing plans data

`import {Block} from "payload/types";
import {invertBackground} from "../../fields/invertBackground";
import {paddingRow, spacingOptions} from "../../fields/padding";

const PricingBlock: Block = {
    fields: [
        invertBackground,
        paddingRow,
        {
            name: 'gutter',
            label: 'Gutter',
            type: 'select',
            options: spacingOptions,
        },
        {
            type: 'array',
            name: 'plans',
            fields: [
                {
                    type: 'row',
                    fields: [
                        {
                            name: 'name',
                            label: 'Name',
                            type: 'text',
                        },
                        {
                            name: 'price',
                            label: 'Price',
                            type: 'text',
                        },
                    ]
                },
                {
                    name: 'description',
                    type: 'textarea',
                },
                {
                    name: 'features',
                    label: 'Features',
                    type: 'array',
                    minRows: 1,
                    fields: [
                        {
                            name: 'feature',
                            label: 'Feature',
                            type: 'text',
                        },
                    ],
                },
            ],
        },
    ],
    slug: 'pricingBlock',
}

export default PricingBlock;
`

I get the same error

`[22:04:26] ERROR (payload): There was an error while saving a version for the Page with ID 7.
    [22:04:26] ERROR (payload): null value in column "_parent_id" of relation "_pages_v_blocks_pricing_block_plans_features" violates not-null constraint
err: {
    "type": "DatabaseError",
        "message": "null value in column \"_parent_id\" of relation \"_pages_v_blocks_pricing_block_plans_features\" violates not-null constraint",
        "stack":
    error: null value in column "_parent_id" of relation "_pages_v_blocks_pricing_block_plans_features" violates not-null constraint
    at /Users/me/Sites/Futto/cms-futto/node_modules/pg/lib/client.js:526:17
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at /Users/me/Sites/Futto/cms-futto/node_modules/src/node-postgres/session.ts:64:19
    at insertArrays (/Users/me/Sites/Futto/cms-futto/node_modules/@payloadcms/db-postgres/src/upsertRow/insertArrays.ts:65:7)
    at insertArrays (/Users/me/Sites/Futto/cms-futto/node_modules/@payloadcms/db-postgres/src/upsertRow/insertArrays.ts:75:7)
    at upsertRow (/Users/me/Sites/Futto/cms-futto/node_modules/@payloadcms/db-postgres/src/upsertRow/index.ts:235:7)
    at Object.createVersion (/Users/me/Sites/Futto/cms-futto/node_modules/@payloadcms/db-postgres/src/createVersion.ts:27:18)
    at saveVersion (/Users/me/Sites/Futto/cms-futto/node_modules/payload/src/versions/saveVersion.ts:100:18)
    at updateByID (/Users/me/Sites/Futto/cms-futto/node_modules/payload/src/collections/operations/updateByID.ts:276:16)
    at updateByIDHandler (/Users/me/Sites/Futto/cms-futto/node_modules/payload/src/collections/requestHandlers/updateByID.ts:36:17)
    "length": 318,
        "name": "error",
        "severity": "ERROR",
        "code": "23502",
        "detail": "Failing row contains (1, null, 1, test, 65626f65515f57e21b8f22c8).",
        "schema": "public",
        "table": "_pages_v_blocks_pricing_block_plans_features",
        "column": "_parent_id",
        "file": "execMain.c",
        "line": "1883",
        "routine": "ExecConstraints"
}`

**Comment by danny1ng** [comment]
> Hi
> 
> I face exactly the same issue for the same use case (pricing block and pricing/plans data)
> 
> Is there a workaround for this ?

Hi. 
I find a solution. If you need the 3rd depth, use the block field in the array field, use the array field in the block field

**Comment by ixoye36** [comment]
Thanks @danny1ng 
That worked

HAs anyone tested the fix ? It doesn't seem to work on my end, I still get the error.

Here is my config

```typescript
import {Block} from "payload/types";
import {invertBackground} from "../../fields/invertBackground";
import {paddingRow, spacingOptions} from "../../fields/padding";
import {PricingFeatureItem} from "../PricingFeatureItem";

const PricingSection: Block = {
    fields: [
        invertBackground,
        paddingRow,
        {
            name: 'gutter',
            label: 'Gutter',
            type: 'select',
            options: spacingOptions,
        },
        {
            type: 'array',
            name: 'plans',
            fields: [
                {
                    type: 'row',
                    fields: [
                        {
                            name: 'name',
                            label: 'Name',
                            type: 'text',
                        },
                        {
                            name: 'price',
                            label: 'Price',
                            type: 'number',
                        },
                        {
                            name: 'buttonText',
                            label: 'Button Text',
                            type: 'text',
                        }
                    ]
                },
                {
                    name: 'description',
                    type: 'textarea',
                },
                /* { // This is the workaround provided by danny1ng
                    name: 'features',
                    label: 'Features',
                    type: 'blocks',
                    minRows: 1,
                    maxRows: 1,
                    blocks: [PricingFeatureItem],
                }, */
                {
                    name: 'list', // this doesn't work
                    type: 'array',
                    fields: [
                        {
                            name: 'feature',
                            type: 'text',
                            label: 'Feature',
                        },
                    ],
                },
            ],
        },
    ],
    slug: 'pricingSection',
}

export default PricingSection;




```

**Comment by DanRibbens** [comment]
I can't reproduce the problem on latest Payload.

Here is a screenshot of my network tab where I've saved with a config that matches your example of block => array => array

![image](https://github.com/payloadcms/payload/assets/6434612/319bf36a-9d73-4653-ba6a-9e3003909bd7)

Block code: 
```ts
{
      name: 'blocksWithArray',
      blocks: [
        {
          fields: [
            {
              name: 'plans',
              fields: [
                {
                  fields: [
                    {
                      name: 'name',
                      type: 'text',
                    },
                    {
                      name: 'price',
                      type: 'number',
                    },
                    {
                      name: 'buttonText',
                      type: 'text',
                    },
                  ],
                  type: 'row',
                },
                {
                  name: 'description',
                  type: 'textarea',
                },
                {
                  name: 'list', // this doesn't work
                  fields: [
                    {
                      name: 'feature',
                      type: 'text',
                    },
                  ],
                  type: 'array',
                },
              ],
              type: 'array',
            },
          ],
          slug: 'array',
        },
      ],
      type: 'blocks',
    },
```
If anyone has more info I think it would be best to open a new issue with a new reproduction considering the original one is working as far ass I can tell.
